### PR TITLE
Fixes to existing infrastructure to get emulator up and running 

### PR DESCRIPTION
**Pull request by Nashatyrev**
This is the first phase of creating emulator: several validator services which interact through a local Wire stub. 
In this phase I will be collecting fixes and refactorings to existing classes

### CODE REVIEW

File: chain/src/test/java/org/ethereum/beacon/chain/StreamTests.java [hunk 1]
```diff
@@ -43,4 +42,25 @@ public void test1() throws InterruptedException {
 
     Thread.sleep(1000L);
   }
+
+  @Test
+  @Ignore
+  public void intervalTest1() throws InterruptedException {
```
Comment (zilm13) [inline chain/src/test/java/org/ethereum/beacon/chain/StreamTests.java]: What are you testing here?

File: chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java [hunk 2]
```diff
@@ -53,32 +52,35 @@
   private final Publisher<BeaconTuple> beaconPublisher;
 
   private static final int UPDATE_MILLIS = 500;
-  private ScheduledExecutorService regularJobExecutor;
-  private ScheduledExecutorService continuousJobExecutor;
+  private Scheduler regularJobExecutor =
```
Comment (zilm13) [inline chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java]: is it lazy? we don't need both until we need this service started

File: chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java [hunk 3]
```diff
@@ -103,29 +105,22 @@ public ObservableStateProcessorImpl(
...
+    Flux.from(slotTicker).subscribe(this::onNewSlot);
+    Flux.from(attestationPublisher).subscribe(this::onNewAttestation);
+    Flux.from(beaconPublisher).subscribe(this::onNewBlockTuple);
+    regularJobExecutor.executeAtFixedRate(
+        Duration.ZERO, Duration.ofMillis(UPDATE_MILLIS), this::doHardWork);
   }
 
   private void runTaskInSeparateThread(Runnable task) {
-    continuousJobExecutor.execute(task);
+    continuousJobExecutor.execute(() -> {
```
Comment (zilm13) [inline chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java]: maybe add `execute(Runnable task, Consumer<Exception> failCallback)`?

File: validator/src/main/java/org/ethereum/beacon/validator/BeaconChainValidator.java [hunk 4]
```diff
@@ -294,11 +307,17 @@ private boolean isInitialized() {
   }
 
   /* FIXME: stub for streams. */
```
Comment (mkalinin) [inline validator/src/main/java/org/ethereum/beacon/validator/BeaconChainValidator.java]: Let's remove this comment as it looks outdated.

File: chain/src/test/java/org/ethereum/beacon/chain/StreamTests.java (hunk 1 above)
Comment (Nashatyrev) [reply chain/src/test/java/org/ethereum/beacon/chain/StreamTests.java]: Just some experiments. Actually `SlotTicker` should consist just of several lines, but `Flux.interval` has missing feature: it accounts `initialDelay` only on subscribing. Will be working on this

File: chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java (hunk 3 above)
Comment (Nashatyrev) [reply chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java]: This is just a temp solution. You may always handle returned `CompletableFuture`, but I'm going to make generic solution for all executors to log unhandled exceptions

File: chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java (hunk 2 above)
Comment (Nashatyrev) [reply chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java]: Nope :/ 
Moved them back to `start()`

**Pull request merged**